snakemake/ioutils/branch.py

```
@@ -0,0 +1,65 @@
+
+    def do_branch_then_otherwise(wildcards):
+        if handle_callable(condition, wildcards):
+            return handle_callable(then, wildcards)
+        else:
+            return handle_callable(otherwise, wildcards)
+
+    def do_branch_cases(wildcards):
+        res = handle_callable(condition, wildcards)
+        try:
+            selected_case = cases[res]
+        except KeyError as e:
```

Ruff (0.8.2)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)